packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/NamedTypeSymbolProvider.cs

```diff
@@ -286,14 +285,11 @@ private CSharpType GetCSharpType(ITypeSymbol typeSymbol)
             Type? type = System.Type.GetType(fullyQualifiedName);
             if (type is null)
             {
-                if (typeSymbol.TypeKind == TypeKind.Error)
-                    throw new InvalidOperationException($"Unable to convert ITypeSymbol: {fullyQualifiedName} to a CSharpType in {Name}");
-
                 return ConstructCSharpTypeFromSymbol(typeSymbol, fullyQualifiedName, namedTypeSymbol);
```

Comment:
Since we now will fall back to constructing from the symbol, can we remove the if block that checks if the name starts with the RootNamespace?

Reply:
Nevermind, I don't think we want to do this as it will then attempt to look up generated types using GetType which would actually succeed if a custom partial classes are added.